src/__tests__/split-at.test.js

```diff
@@ -0,0 +1,105 @@
+/* eslint-env node, jest */
+const { splitAt, range, slice, asyncSplitAt, asyncToArray } = require('..')
+
+const u = undefined
+
+describe('splitAt', function () {
+  it('works when the halves are consumed in order', function () {
+    const [[a, b, c], [d, e, f]] = splitAt(3, range())
+    expect([[a, b, c], [d, e, f]]).toEqual([[0, 1, 2], [3, 4, 5]])
+  })
+
+  it('works when the source is exhuasted while the first half is being consumed', function () {
+    const [[a, b, c], [d, e, f]] = splitAt(3, slice(2, range()))
+    expect([[a, b, c], [d, e, f]]).toEqual([[0, 1, u], [u, u, u]])
+  })
+
+  it('works when the source is exhuasted while the second half is being consumed', function () {
+    const [[a, b, c], [d, e, f]] = splitAt(3, slice(4, range()))
+    expect([[a, b, c], [d, e, f]]).toEqual([[0, 1, 2], [3, u, u]])
```

Comment:
Why don't you just use `undefined`? Using `u` alias may make the code looks pretty (which is a total coincidence), but it also makes the code harder to read.

Reply:
Haha what do you mean by "a total coincidence"? The point was to make it look nice. I think it is fairly self-evident from the string description, if nothing else, what `u` means. I can change it back to undefined too though, I do not really care.

Reply:
Well, I guess I should've said "a mere coincidence" instead (I'm not a native English speaker).

> The point was to make it look nice.

I already knew that, and "nice" is not a good reason.

Reply:
To clarify further, the "coincidence" that I said is not "it happens to be pretty" but "the numbers  happen to have 1 character long the array happens to have 3 elements so you can use `u` to make it pretty"

Reply:
I don't think you mean coincidence at all. If it was a coincidence that would mean that I did not intend to make it look nice.

Reply:
Ah OK I see what you mean. Yeah I know that it only looks nice for those reasons, but we do most of our tests with single digit numbers. But anyway, I'm not really meaning to make a debate out of this. I'll change it back.